### PR TITLE
LineCanvas support for mixing double and single lines

### PR DESCRIPTION
**Pull request by tznind**
Fixes #_____ - Include a terse summary of the change or which issue is fixed.

## Pull Request checklist:

- [ ] I've named my PR in the form of "Fixes #issue. Terse description."
- [ ] My code follows the [style guidelines of Terminal.Gui](https://github.com/gui-cs/Terminal.Gui/blob/develop/.editorconfig) - if you use Visual Studio, hit `CTRL-K-D` to automatically reformat your files before committing.
- [ ] My code follows the [Terminal.Gui library design guidelines](https://github.com/gui-cs/Terminal.Gui/blob/develop/CONTRIBUTING.md)
- [x] I ran `dotnet test` before commit
- [ ] I have made corresponding changes to the API documentation (using `///` style comments)
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any poor grammar or misspellings
- [x] I conducted basic QA to assure all features are working

### CODE REVIEW

**Comment by tznind** [comment]
Mixed line style support, have doubles merge into singles and vice versa!

On my laptop on holiday so can't do the Ctrl+K+D thing but otherwise should be good to go.

File: UnitTests/LineCanvasTests.cs [hunk 1]
```diff
@@ -218,6 +218,68 @@ public void TestLineCanvas_Window_Double ()
...
+
+
+			canvas.AddLine (new Point (5, 0), 4, Orientation.Vertical,thinStyle);
+			canvas.AddLine (new Point (0, 2), 9, Orientation.Horizontal, BorderStyle.Double);
+
+			v.Redraw (v.Bounds);
+
+			string looksLike =
+@"    
+╒════╤═══╕
```
Comment (tig) [inline UnitTests/LineCanvasTests.cs:243]: I don't see "DoubleTop" here??

Comment (tig) [reply UnitTests/LineCanvasTests.cs:243]: Yea, seems not all orientations are working:

image

**Review by tig — CHANGES_REQUESTED** [review]
See comments. It's not working for all orientations.

**Comment by tig** [comment]
Also, @tznind, can you please move `LineCanvasTests.cs` to the `Core` folder and ensure the namespace is `Terminal.Gui.Core`?

File: UnitTests/LineCanvasTests.cs (hunk 1 above)
Comment (tznind) [reply UnitTests/LineCanvasTests.cs:243]: Hmn weird... Did you definitely `git pull` the latest? Should be a443ec7?

I did the corners first and only got to the Ts in a443ec7 so this behavior would be consistent with the older commit f716dbe596c7086754278aba54e36d3a63b199b4 only.

Comment (tznind) [reply UnitTests/LineCanvasTests.cs:243]: shot-2023-02-11_08-11-04

**Comment by tznind** [comment]
> Also, @tznind, can you please move `LineCanvasTests.cs` to the `Core` folder and ensure the namespace is `Terminal.Gui.Core`?

Done.  Guessing you mean `Terminal.Gui.CoreTests` though right? Not just`Terminal.Gui.Core`?

File: UnitTests/LineCanvasTests.cs (hunk 1 above)
Comment (tig) [reply UnitTests/LineCanvasTests.cs:243]: I did

image

This is a direct link to the test that seems incorrect in this PR:

https://github.com/tznind/gui.cs/blob/d555576a138d222313c0efe8f35c2149eed196a6/UnitTests/Core/LineCanvasTests.cs#L243

This is what i see when look at your fork with this branch:

image

**Comment by tig** [comment]
I just grabbed your PR again using a different technique and now it's correct and working correctly.

```
git remote add tznind tig:tznind/gui.cs.git
git fetch tznind
git checkout -b tznind-line-canvas-style-mixing tznind/line-canvas-style-mixing  
```

😕

**Review by tig — APPROVED** [review]
<3

**Comment by tznind** [comment]
@tig just recording this here as I don't think theres much we can do about it but the current implementation still looks a bit janky when changing from double to single 'mid rune' along a specific axis.  For example if you draw 2 boxes, one with single width and one with double width overlapping:

image

**Pull request merged**